R2API/ItemAPI.cs

```diff
@@ -112,12 +113,23 @@ public static class ItemAPI {
                 R2API.Logger.LogError($"Too late ! Tried to add item: {item.ItemDef.nameToken} after the item list was created");
             }
 
-            if (item.ItemDef == null) {
+            if (!item.ItemDef) {
                 R2API.Logger.LogError("Your ItemDef is null ! Can't add your item.");
             }
 
             if (string.IsNullOrEmpty(item.ItemDef.name)) {
                 R2API.Logger.LogError("Your ItemDef.name is null or empty ! Can't add your item.");
+            }
+
+            if (!item.ItemDef.pickupModelPrefab) {
+                R2API.Logger.LogWarning($"No ItemDef.pickupModelPrefab, the game will show nothing when the item is on the ground.");
+            }
+            else if (item.ItemDisplayRules != null &&
+                item.ItemDisplayRules.Dictionary.Values.Any(rules => rules.Any(rule => rule.ruleType == ItemDisplayRuleType.ParentedPrefab)) &&
+                !item.ItemDef.pickupModelPrefab.GetComponent<ItemDisplay>()) {
+                R2API.Logger.LogWarning($"Your ItemDef.pickupModelPrefab ({item.ItemDef.name}) does not have an ItemDisplay component attached to it " +
+                    $"(you have ItemDisplayRuleType.ParentedPrefab rules), " +
+                    $"the pickup model should have one and have atleast a rendererInfo in it for having correct visibility levels.");
```

Comment:
No need for formatting literal

Reply:
having to scroll vertical bar is annoying though

Reply:
I meant more that the last line doesn't need the $

Reply:
oh sorry misunderstood